lib/rules/no-classname-on-common-components.js

```diff
@@ -0,0 +1,21 @@
+"use strict";
+var getProp = require('jsx-ast-utils/getProp');
+
+module.exports = function(context) {
+    var options = context.options[0] || {components: []};
+    return {
+        JSXElement: function(node) {
+            var name = node.openingElement.name.name
+            if (options.components.indexOf(name) === -1) {
+                return;
+            }
+            var typeProp = getProp(node.openingElement.attributes, 'className');
+            if (!typeProp) {
+                return;
+            }
+            context.report(node, 'Avoid className prop on common component ' + name);
```

Comment:
For future reference, this will _probably_ be running in at least a Node 4+ environment (and it will for sure be in our codebase), so it should be safe to use some basic ESNext features, such as `const`, as well as [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) for string concat here.